---
bool success
float32 width
